plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/frontend.ts

```diff
@@ -169,6 +170,74 @@ const addToCartWithOptionsStore = store(
 				);
 				return matchedVariation?.variation_id || null;
 			},
+			get allowsDecrease() {
+				const context = getContext< Context >();
+				const { quantity, childProductId, productType } = context;
+
+				const currentQuantity =
+					productType === 'grouped' && childProductId
+						? quantity[ childProductId ] || 0
+						: quantity[ context.productId ] || 0;
+
+				const selector =
+					productType === 'grouped' && childProductId
+						? `.wc-block-components-quantity-selector__input[name="quantity[${ childProductId }]"]`
+						: `.wc-block-components-quantity-selector__input[name="quantity"]`;
+
+				const inputElement = document.querySelector(
+					selector
+				) as HTMLInputElement | null;
+
+				if ( ! inputElement ) {
+					return false;
+				}
+
+				const parsedMinValue = parseInt( inputElement.min, 10 );
+				const parsedStep = parseInt( inputElement.step, 10 );
+
+				// For grouped product children, allow quantity of 0.
+				const defaultMinValue = childProductId ? 0 : 1;
+				const minValue = isNaN( parsedMinValue )
+					? defaultMinValue
+					: parsedMinValue;
+				const step = isNaN( parsedStep ) ? 1 : parsedStep;
+
+				return currentQuantity - step >= minValue;
+			},
+			get allowsIncrease() {
+				const context = getContext< Context >();
+				const { quantity, childProductId, productType } = context;
+
+				const currentQuantity =
+					productType === 'grouped' && childProductId
+						? quantity[ childProductId ] || 0
+						: quantity[ context.productId ] || 0;
+
+				const selector =
+					productType === 'grouped' && childProductId
+						? `.wc-block-components-quantity-selector__input[name="quantity[${ childProductId }]"]`
+						: `.wc-block-components-quantity-selector__input[name="quantity"]`;
+
+				const inputElement = document.querySelector(
+					selector
+				) as HTMLInputElement | null;
+
+				if ( ! inputElement ) {
+					return false;
+				}
+
+				const parsedMaxValue = parseInt( inputElement.max, 10 );
+				const parsedStep = parseInt( inputElement.step, 10 );
+
+				const maxValue = isNaN( parsedMaxValue )
+					? undefined
+					: parsedMaxValue;
+				const step = isNaN( parsedStep ) ? 1 : parsedStep;
+
+				return (
+					maxValue === undefined || currentQuantity + step <= maxValue
+				);
```

Comment:
Small nit, but I think using an early return when `parsedMaxValue` is NaN would make it easier to read:

```suggestion
				const parsedMaxValue = parseInt( inputElement.max, 10 );

				if ( isNaN( parsedMaxValue ) ) {
					return true;
				}
				
				const parsedStep = parseInt( inputElement.step, 10 );
				const step = isNaN( parsedStep ) ? 1 : parsedStep;

				return currentQuantity + step <= maxValue;
```

This is quite subjective, so I'm fine leaving it as-is if you prefer. :slightly_smiling_face: In case you end up changing this, it can be changed in the `add-to-cart-form/frontend.ts` file as well.